core/src/main/java/org/apache/spark/util/collection/unsafe/sort/UnsafeSorterSpillWriter.java

```diff
@@ -48,7 +48,7 @@
   private final File file;
   private final BlockId blockId;
   private final int numRecordsToWrite;
-  private DiskBlockObjectWriter writer;
+  private final DiskBlockObjectWriter writer;
```

Comment:
I think that this was purposely non-final so that we can null it out in order to allow the writer to be garbage collected as soon as `close()` is called.